src/BSeries.jl

```diff
@@ -1453,4 +1456,379 @@ include("latexify.jl")
+    branchs = get_branchs(a)
+    #we obtain the number of branchs the right-most trunk has
+    #the Theorem 2 in the article requires to know if m is odd or even
+    m = num_branchs(a)
+    #we create an array from 1 to m plus another node for the final branch of the rightmost trunk
+    #energy_preserving_partner
+    energy_preserving_partner = collect(1:m+1)
+    #then, we insert every level_sequence from branchs
+    for j in 1:m
+        last_j_occurrence = findlast(x -> x == j, energy_preserving_partner)
+        last_jplus1_occurrence = findlast(x -> x == j+1, energy_preserving_partner)
+        energy_preserving_partner = vcat(energy_preserving_partner[1:last_j_occurrence], branchs[j], energy_preserving_partner[last_jplus1_occurrence:end])
+    end
+    energy_preserving_partner = canonicalarray(energy_preserving_partner)
+    return energy_preserving_partner
+end
+
+
+#This function rearranges the level sequence in 
+#the same way as the Bseries output does
+
+function canonicalarray(tree)
+    t = rootedtree(tree)
+    trees = t.level_sequence
+    return trees
+end
+
+#This function generates all the equivalent trees 
+#for a given level_sequence
+
+function equivalent_trees(array)
+    tree = rootedtree(array)
+    l = length(array)
+    #we get the permutations of the array
+    superarray = collect(permutations(array))
+    lperm = length(superarray)
+    #eliminate all the arrays except for those for which the first element is equal to 1
+    for i in reverse(1:lperm) 
+        if superarray[i][1] != 1
+            splice!(superarray, i)
+        end
+    end
+    lperm = length(superarray)
+    #now, check if the levelsequence is the level sequence represents a RootedTree:
+    #the condition is that each number m cannot be followed by a number n such that
+    #n-m > 1. If that's the case, eliminate this array with 'splice!'
+    RTree_array = true
+    for i in reverse(1:lperm)
+        for j in 1:(l-1)
+            if (superarray[i][j+1] - superarray[i][j]) > 1
+                RTree_array = false
+            end
+        end
+        if RTree_array == false
+            splice!(superarray, i)
+        end
+        RTree_array = true
+    end
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        onetree = rootedtree(superarray[i])
+        isthesame_tree = onetree == tree
+        if isthesame_tree == false
+            splice!(superarray, i)
+        end
+    end
+    repeated_tree = false
+    lperm = length(superarray)
+    #we delete any repeated tree
+    for i in reverse(1:lperm)
+        for j in 1:(i-1)
+            if superarray[i] == superarray[j]
+                repeated_tree = true
+            end
+        end 
+        if repeated_tree == true
+            splice!(superarray, i)
+        end
+        repeated_tree = false
+    end
+
+    return superarray
+end
+
+
+function num_branchs(a)
+    k = a[end]
+    return k - 1
+end
+
+function add_one_to_left(arrays)
```

Comment:
Please add a description for this method.

Reply:
Implemented.